### DISCUSSION TITLE
Fastapi v0.100.0 provides less descriptive error message for schema validation errors

### DISCUSSION THREAD

**Discussion by Harry-Lees** [fastapi/fastapi#10013] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class RootSchema(BaseModel):
    foo: str

@app.get("/", response_model=RootSchema)
def get_root():
    return {}
```

### Description

The above code returns a 500 (Internal Server Error) due to pydantic not being able to validate the returned schema.

On FastAPI version 0.99.1, the error message looked like the following:

```
INFO:     127.0.0.1:58889 - "GET / HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
<Traceback elided for brevity>
  File "/Users/harrylees/Documents/fastapi-test/venv/lib/python3.10/site-packages/fastapi/routing.py", line 145, in serialize_response
    raise ValidationError(errors, field.type_)
pydantic.error_wrappers.ValidationError: 1 validation error for RootSchema
response -> foo
  field required (type=value_error.missing)
```

Which was useful for debugging as it returned the validation errors from Pydantic. On FastAPI v0.100.0 and above, the error message looks like the following:

```
INFO:     127.0.0.1:58970 - "GET / HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
<Traceback elided for brevity>
  File "/Users/harrylees/Documents/fastapi-test/venv/lib/python3.10/site-packages/fastapi/routing.py", line 154, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError
```

In my opinion, this error message is less descriptive and harder to debug.

### Operating System

macOS

### FastAPI Version

0.100.0

### Pydantic Version

2.1.1

### Python Version

3.10.5

### Additional Context

This was originally raised here https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6638673

**Comment by HuiDBK** [comment]
Yes, I also encountered this problem, too little error information was reported when the respose_model was not in line, and the developer was not very good at locating the problem. I always commented out the respose_model to troubleshoot the problem, but it was still a little troublesome

**Comment by Harry-Lees** [comment]
This issue was resolved in https://github.com/tiangolo/fastapi/releases/tag/0.101.1. The error message now returns the following:

```
<entire Traceback elided for brevity>
  File "/Users/harrylees/Documents/GitHub/fastapi/fastapi/routing.py", line 154, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 2 validation errors:
  {'loc': ('response', 'bar'), 'msg': 'extra fields not permitted', 'type': 'value_error.extra'}
  {'loc': ('response', 'baz'), 'msg': 'extra fields not permitted', 'type': 'value_error.extra'}
```
